gevent/_socket3.py

```diff
@@ -22,6 +23,8 @@
 
 
 def _get_memory(string, offset):
+    if isinstance(string,text_type):
+        string = string.encode('utf-8')
```

Comment:
Python 3 doesn't support sending (unicode) strings across a socket, and I don't think gevent should introduce a gratuitous difference here. Instead, we should update the examples to add the appropriate `b` prefixes.

```
Python 3.4.3 (default, May 25 2015, 18:48:21)
[GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.56)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import socket
>>> s = socket.socket()
>>> s.connect(('www.yahoo.com', 80))
>>> s.sendall('GET / HTTP/1.0\r\n\r\n')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'str' does not support the buffer interface
```